processor/src/test/java/com/joom/lightsaber/processor/end2end/End2EndTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.joom.lightsaber.processor.end2end
+
+import com.joom.lightsaber.processor.ErrorReporter
+import com.joom.lightsaber.processor.ErrorReporterImpl
+import com.joom.lightsaber.processor.LightsaberParameters
+import com.joom.lightsaber.processor.LightsaberProcessor
+import java.io.File
+import java.net.URI
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+import kotlin.io.path.extension
+import kotlin.io.path.isDirectory
+import kotlin.streams.toList
+import org.jetbrains.kotlin.cli.common.ExitCode
+import org.jetbrains.kotlin.cli.jvm.K2JVMCompiler
+
+open class End2EndTest(private val root: Path) {
+
+  private val compiledFilesDirectory = root.resolve("_generated")
+  private val processedDirectory = root.resolve("_processed")
+  private val classpath = System.getProperty("java.class.path")
+  private val classpathWithRuntime = computeClasspath(classpath)
+  private val classpathWithAllChildJars = filterDirectoriesAndJars(classpathWithRuntime)
+
+  protected fun processProject(
+    sourceCodeDir: String,
+    errorReporter: ErrorReporter = ErrorReporterImpl(),
+    ignoreErrors: Boolean = false
+  ) {
+    val compiled = compile(root.fileName.toString() + File.separator + sourceCodeDir)
+    val parameters = LightsaberParameters(
+      inputs = listOf(compiled),
+      outputs = listOf(processedDirectory),
+      bootClasspath = listOf(Paths.get(URI.create("jrt:/")).resolve("/modules/java.base")),
+      classpath = classpathWithAllChildJars,
+      projectName = sourceCodeDir,
+      gen = processedDirectory,
+      errorReporter = errorReporter
+    )
+
+    try {
+      LightsaberProcessor(parameters).process()
+    } catch (e: Throwable) {
+      if (!ignoreErrors) throw e
+    } finally {
+      cleanDir(compiledFilesDirectory)
+      cleanDir(processedDirectory)
+    }
+  }
+
+  private fun compile(sourceCodeDir: String): Path {
+    cleanDir(compiledFilesDirectory)
+
+    val input = this::class.java.classLoader.getResource(sourceCodeDir)?.toAbsolutePath()
+      ?: throw IllegalArgumentException("Cannot find directory $sourceCodeDir in test resources")
+
+    val compiler = K2JVMCompiler()
+    // https://kotlinlang.org/docs/compiler-reference.html#common-options
+
+    val exitCode = compiler.exec(
+      System.err,
+      input,
+      "-d", compiledFilesDirectory.absolutePathString(),
+      "-cp", classpath,
+      "-jvm-target", "1.8",
```

Comment:
Shouldn't it be 9 or later with the new version of Grip?

Reply:
it's experimental params, removed